src/NumericalAlgorithms/LinearOperators/PowerMonitors.cpp

```diff
@@ -233,76 +235,49 @@ double convergence_rate(const DataVector& power_monitor,
     denom += one_over_denom_this_term;
     num += gsl::at(slopes, i) * one_over_denom_this_term;
   }
-  return -num / denom;
-}
+  result.convergence_rate = -num / denom;
 
-double number_of_pile_up_modes(const DataVector& power_monitor,
-                               const size_t number_of_filtered_modes) {
-  // Need enough unfiltered modes to compute the convergence rate. Here,
-  // require at least 4 unfiltered modes.
-  ASSERT(
-      power_monitor.size() > number_of_filtered_modes + 3,
-      "Power monitor needs at least 4 unfiltered modes to compute convergence "
-      "rate and number of pile up modes, but power monitor has size "
-          << power_monitor.size() << " with " << number_of_filtered_modes
-          << " filtered modes");
-  const double conv_rate =
-      convergence_rate(power_monitor, number_of_filtered_modes);
-  // If the convergence rate is nearly zero, return zero pile up modes.
+  // Compute number of pile up modes
+  // First, if the convergence rate is nearly zero, return zero pile up modes.
   // This avoids dividing by zero, and is reasonable in the sense that if
   // the convergence rate is very small, it's not realistic to distinguish
   // whatever convergence you might have left and pile up modes.
```

Comment:
I realize this is an old comment, but maybe giving an example would be good? E.g. "This means that the function is not at all resolved, eg a step function, and so the power spectrum is approximately flat." I know you're saying pretty much the same thing but it took me a bit of thinking to figure it out, and I've thought about this a lot.